core/src/main/java/org/jboss/jandex/IndexReaderV2.java

```diff
@@ -316,7 +318,8 @@ private AnnotationInstance readAnnotationEntry(PackedDataInputStream stream, Ann
         DotName name = nameTable[stream.readPackedU32()];
         AnnotationTarget target = readAnnotationTarget(stream, caller);
         AnnotationValue[] values = readAnnotationValues(stream);
-        return new AnnotationInstance(name, target, values);
+        boolean visible = (version >= 11) && stream.readBoolean();
```

Comment:
This will be `false` for `versions <= 10`, right? That would be wrong. Older indicies only contain runtime-visible annotations, so this should be `true` for `versions <= 10`. I'd write it like this:

```java
boolean visible = true;
if (version >= 11) {
    visible = stream.readBoolean();
}
```

Like this, it's identical to other places in the code that decide based on version.

Reply:
Well, that's embarrassing... I'll fix it.

Reply:
Fixed this. It turns out that the `IndexWriterV2` required some additional changes also to avoid writing the invisible annotations entirely for versions less than 11.